sky/skylet/log_lib.py

```diff
@@ -440,6 +447,14 @@ def tail_logs(job_id: Optional[int],
         with open(log_path, 'r', newline='', encoding='utf-8') as log_file:
             # Using `_follow` instead of `tail -f` to streaming the whole
             # log and creating a new process for tail.
+            if tail > 0:
+                lines = collections.deque(log_file.readlines(), maxlen=tail)
+                for line in lines:
+                    print(line, end='')
+                # Flush the last n lines
+                print(end='', flush=True)
```

Comment:
If the `start_stream_at` is within the last `tail` lines, this will show all the lines before the `start_stream_at`?

Reply:
I wasn't aware that there is also a `start_stream_at` check here. I will change it to use the same logic as above.